Lib/Inc/CUartCom.h

```diff
@@ -11,25 +11,39 @@
 #ifndef CUARTCOM_H_
 #define CUARTCOM_H_
 
+#include <queue>
 #include <string>
+#include "CCmdBuffer.h"
+#include "CGpioWrapper.h"
+#include "IComChannel.h"
 #include "usart.h"
 
-#define UART_TIMEOUT 100
-
-class CUartCom
+#define UART_TIMEOUT   100
+#define MAX_QUEUE_SIZE 20
+class CUartCom : public IComChannel
 {
-private:
-    UART_HandleTypeDef *mp_huart;
-    GPIO_TypeDef *m_uart_de_port = nullptr;
-    uint16_t m_uart_de_pin;
-
 public:
-    CUartCom(UART_HandleTypeDef *p_huart);
+    CUartCom(UART_HandleTypeDef *p_huart, const std::string name);
     CUartCom(UART_HandleTypeDef *p_huart,
+             const std::string name,
              GPIO_TypeDef *uart_de_port,
              uint16_t uart_de_pin);
 
-    void sendMessage(const std::string &msg);
+    void startReception();
```

Comment:
This is perhaps can be named better. The name of the method does not quite correspond with what you are doing inside. Most important thing you are doing is initialising the static variables. Fact you also enable interrupt is not really that important.